exporter/kinesisexporter/exporter_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kinesisexporter
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+type producerMock struct {
+	mock.Mock
+}
+
+func (m *producerMock) start() {
+	m.Called()
+}
+
+func (m *producerMock) stop() {
+	m.Called()
+}
+
+func (m *producerMock) put(data []byte, partitionKey string) error {
+	args := m.Called(data, partitionKey)
+	return args.Error(0)
+}
+
+func TestNewKinesisExporter(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	require.NotNil(t, cfg)
+
+	exp, err := newExporter(cfg, zap.NewNop())
+	assert.NotNil(t, exp)
+	assert.NoError(t, err)
+}
+
+func TestNewKinesisExporterBadEncoding(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	require.NotNil(t, cfg)
+	cfg.Encoding = ""
+
+	exp, err := newExporter(cfg, zap.NewNop())
+	assert.Nil(t, exp)
+	assert.Error(t, err)
+	assert.Equal(t, err.Error(), "unrecognized encoding")
```

Comment:
You can use https://godoc.org/github.com/stretchr/testify/assert#Errorf to be more obvious as to what you're trying to test here.